internal/config/tokens.go

```diff
@@ -181,11 +182,18 @@ func keepConfigTokensFresh(ctx context.Context, m *sync.Mutex, t *tokens.Tokens,
 // the user's browser. Set the UserURLCallback package variable if you want to
 // support this.
 //
-// Don't call this when other goroutines might also be accessing t.
-func refreshDischargeTokens(ctx context.Context, t *tokens.Tokens, uucb UserURLCallback) (bool, error) {
-	updateOpts := []tokens.UpdateOption{tokens.WithDebugger(logger.FromContext(ctx))}
+// Don't call this when other goroutines might also be accessing it.
+func refreshDischargeTokens(ctx context.Context, t *tokens.Tokens, uucb UserURLCallback, advancePrune time.Duration) (bool, error) {
+	updateOpts := []tokens.UpdateOption{
+		tokens.WithDebugger(logger.FromContext(ctx)),
+		tokens.WithAdvancePrune(advancePrune),
+	}
 
 	if uucb != nil {
+		updated, err := t.Update(ctx, updateOpts...)
+		if err == nil || !strings.Contains(err.Error(), "missing user-url callback") {
+			return updated, err
+		}
```

Comment:
A comment explaining that we can only parallelize if `WithUserURLCallback` is omitted might be useful. Without that knowledge it's confusing why we'd do it this way.